src/BlobField/index.js

```diff
@@ -14,15 +14,52 @@ export default function BlobField({ collapsed = false }) {
   const blobsRef = useRef(null);
   const history = useHistory();
   const [activeArtist, setActiveArtist] = useState(null);
+  const [blobEvent, setBlobEvent] = useState(null);
   const [height, setHeight] = useState(window.innerHeight);
   const [width, setWidth] = useState(window.innerWidth);
   const [parentWidth, setParentWidth] = useState(window.innerWidth);
   const [parentHeight, setParentHeight] = useState(window.innerHeight);
+  const popoverStyle = usePopooverStyle(blobEvent);
+
+  function usePopooverStyle(event) {
+    let style = {
+      display: 'none'
+    };
+    
+    if (collapsed && event && event.type === "mouseenter") {
+      const isVertical = window.innerWidth > window.innerHeight
+      const count = config.artists.length;
+      const totalLength = isVertical ? window.innerHeight : document.body.clientWidth;
+      const canvasLength = totalLength / count;
+
+      let pos = event.target.position;
+      if (isVertical && pos.x < canvasLength){
+        style = {
+          top: pos.y,
+          right: canvasLength + 5,
+          transform: 'translate(0, -50%)'
+        };
+      }
+      else if (!isVertical && pos.y < canvasLength){
+        style = {
+          bottom: canvasLength + 5,
+          left: '50%',
+          transform: 'translate(-50%, 0)'
+        };
+      }
+    }
+    return style;
+  }
 
   useEffect(() => {
     const onArtistHovered = (event) => {
-      if (event.type === "mouseenter") setActiveArtist(event.target.artist);
-      else setActiveArtist(null);
+      if (event.type === "mouseenter") {
+        setActiveArtist(event.target.artist);
+        setBlobEvent(event);
```

Comment:
It would be better to calculate the style here rather than storing the event. With an event that our code does not have control over, it could have weird dependencies leading to a memory leak or change before we use it in the render func leading to unintentional bugs. (setState and it's hook-style spinoffs trigger render asynchronously)
If you calculate style at the time of the callback, you know exactly what you'll get when you rerender.

Reply:
I see what you're saying. But I don't think it really applies here. 

-  Our `event.target` is defined by us in `ball.js`
-  If paper.js changes their path definitions we have to modify our entire blob codebase let alone this part.
- `setSate` merges the sate objects. So everytime an event is passed to `setBlobEvent` it will overwrite the previous one. Given that only hover events are passed to it we can be sure that we maintain the que and always end up with the latest event object and properties.
- Function call stack remains the same although asynchronous:
   `onArtistHovered` --> `setBlobEvent` --> `setPopoverStyle`
   Even if called many times, the last `setPopoverStyle` will take as its parameter the very last `event` object.

All these aside, I don't understand how storing the `event` is really different from storing `event.target.artist` which is what we're doing and later using in our render function `activeArtist.name`. The only difference is that `setPopStyle` is called to modify the style state, and I doubt it takes more than 0.5millisecs to compute. 

Anyways, to keep you happy I just pushed a commit that doesn't store the event. https://github.com/blerchin/dma-20-mfa-show/commit/14f1ce6d6ade326565891104b2aa60245e8ae031

Reply:
I don't think it's actually causing bugs in this case. More of a general react style thing and AFAIK those are the reasons it's preferred. The other one, is that react does an equality check on the argument passed to `setState` to decide whether to rerender. If you were working with a library that reused event objects you definitely wouldn't want to do this. Not the case with paper, but still...
I don't love the pattern where `onArtistHovered` returns an event either, but given that it has multiple attrs we need (including `artist`, which does get provided initially by `BlobField`) I'm not complaining too much.

Anyway, looks good. Thanks!

Reply:
Makes sense. After refactoring I did realize that the whole event state was completely unnecessary, just taking up memory.